### PR TITLE
Download reports from FTP

### PR DESCRIPTION
**Pull request by rkondratowicz**
todo:
- parse csv and set PUT to letters service
- scheduled job
- insights

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -52,6 +52,7 @@ module "consumer" {
     FTP_PORT          = "${var.ftp_port}"
     FTP_FINGERPRINT   = "${var.ftp_fingerprint}"
     FTP_TARGET_FOLDER = "${var.ftp_target_folder}"
+    FTP_TARGET_FOLDER = "${var.ftp_reports_folder}"
```
Comment (doncem) [inline infrastructure/main.tf]: incorrect name

**Review by doncem — CHANGES_REQUESTED** [review]
serious dance of try/returns 🔬

File: src/main/java/uk/gov/hmcts/reform/slc/services/steps/sftpupload/FtpClient.java [hunk 2]
```diff
@@ -77,7 +84,35 @@ public void upload(PdfDoc pdfDoc) {
...
+     */
+    public List<byte[]> downloadReports() {
+        return runWith(sftp -> {
+            try {
+                SFTPFileTransfer transfer = sftp.getFileTransfer();
+
+                return sftp.ls(reportsFolder)
+                    .stream()
+                    .filter(RemoteResourceInfo::isRegularFile)
+                    .map(file -> {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/slc/services/steps/sftpupload/FtpClient.java:98]: Entire lambda could be extracted into a function?

File: infrastructure/main.tf (hunk 1 above)
Comment (rkondratowicz) [reply infrastructure/main.tf]: fixed

**Review by doncem — APPROVED** [review]

**Review by nitinprabhu — APPROVED** [review]

**Pull request merged**